xapian-core/matcher/synonympostlist.h

```diff
@@ -58,6 +58,12 @@ class SynonymPostList : public PostList {
      */
     bool want_unique_terms;
 
+    /** Flag indicating whether the weighting object needs the max wdf of
+     *  terms.
+     *  FIXME : Not implemented
```

Comment:
This really needs to be hooked up - without that either we'll end up calculating this statistic all the time, which is unnecessary overhead when it isn't wanted, or else we'll never actually calculate it, in which case your new code isn't actually used!